### DISCUSSION TITLE
How to click the link after using the system browser to open?

### DISCUSSION THREAD

**Discussion by icewizardry** [wailsapp/wails#1898] [Q&A]
It is not friendly to open a new window in Wails, so I hope to use the system's own browser instead. How to achieve this?

**Comment by misitebao** [comment]
Here are two ways for your reference:

1. You can use the event delegate on the front end to intercept the click event of tag a, and then use the [BrowserOpenURL](https://wails.io/docs/reference/runtime/browser#browseropenurl) method provided by the wails runtime to open the web page:

```javascript
document.body.addEventListener("click", function () {
  var target = event.target || event.srcElement;
  var url = target.getAttribute("href");

  if (target.nodeName.toLocaleLowerCase() === "a") {
    window.runtime.BrowserOpenURL(url);
  }
});
```

2. You can encapsulate a link component, use the link tag where the a tag is needed, and then call the [BrowserOpenURL](https://wails.io/docs/reference/runtime/browser#browseropenurl) method provided by the runtime in the component to open the web page:

```vue
// link.vue
<template>
  <span class="openlink" @click="onClickhandle">
    <slot></slot>
  </span>
</template>

<script>
export default {
  name: "Link",
  props: {
    href: String,
  },
  setup(props) {
    const onClickhandle = () => {
      window.runtime.BrowserOpenURL(props.href);
    };
    return {
      onClickhandle,
    };
  },
};
</script>
```
